momepy/preprocessing.py

```diff
@@ -1067,6 +1068,14 @@ def roundabout_simplification(
     GeoDataFrame
         GeoDataFrame with an updated geometry and an additional column labeling modified edges.
     """
+    ver_str = gpd.__version__
+    ver = re.search("([\d.]).([\d.]+).([\d]+)", ver_str).group(2)
```

Comment:
This method for determining version compatibility seems a bit cumbersome and would produce desired results for version over `v1.0.0` for example. Let try something simpler, like which is already in [here](https://github.com/pysal/momepy/blob/main/momepy/preprocessing.py#L28).

Reply:
that's much cleaner, indeed... 
(slow learnings, here)

Reply:
Don't be hard on yourself! Everything takes some practice!

Reply:
We'll either need to skip tests for that function if version is lower than 0.9.0 ***or*** test that the error is caught if using those versions. The second choice is probably what we want to do. [Here's an example](https://github.com/pysal/momepy/blob/main/tests/test_graph.py#L42) of how to do that.

Reply:
@jGaboardi thanks for that example... I admit I'm not familiar with the WITH statement and therefore will need to get my head around to be able to skip all tests if GDP version is less than 0.9.0 (which by the way is when 'predicate `touches` seems to have been surfaced according to the documentation of 0.8.1 and 0.8.2)

Reply:
So maybe we should add the conditional for less than `0.8.1`?

As for the `with` statement, it is used to isolate a specific block of code. For use in testing it is nice because we can run a test that we know will fail, for example [here](https://github.com/pysal/momepy/blob/main/tests/test_utils.py#L32-L34) we know that a `ValueError` should be raised and we can test for it. In our case, we know that an `ImportError` should be raised and we can test for that *only if* we don't have the proper version of GeoPandas (made `0.8.1`?).

Reply:
sorry what I meant to say is that the first time the predicate 'touches' appears is on 0.9.0 (`touches` is used in a couple of places during the process - so, actually, quite vital for overall success)
[Here are the docs of 0.8.2 ](https://geopandas.org/en/v0.8.2/reference/geopandas.sjoin.html?highlight=sjoin#geopandas.sjoin) for reference.

as for the `with` statement, I think I understand the idea behind but I just don't really know what should be tested (if anything at all) when geopandas is less than 0.9.0... 

my current line 139 : 
`with pytest.raises(TypeError): # unsure how to develop this`

Reply:
Ah, OK. The purpose is to ensure errors are raised when we know they should be raised. And since we know that we need at least version 0.9.0 to use the roundabouts simplification, we should test that an `ImportError` is raised if using less than that (because we added that stipulation into `roundabout_simplification()`. So maybe something the following would be good and then repeat for each of the 5 simplification tests that you wrote:

```python
def test_roundabout_simplification_1(self):
     if not GPD_09:
         with pytest.raises(ImportError):
             mm.roundabout_simplification(self.df_streets_rabs)
     else:
         check = mm.roundabout_simplification(self.df_streets_rabs)
         assert len(check) == 65
         assert len(self.df_streets_rabs) == 88  # checking that nothing has changed
```

And along the same lines, we can also check that the `TypeError` is raised properly by creating a test that passing in MultiLineStrings, Points, etc...

Reply:
OK yeah, that makes more sense in my head now! 
I've used your code for test_1 but I only create an if clause that skips the tests when GPD_09 is false (i.e. less than 0.9.0) for the other 4 tests. 

As for the TypeError, do you mean that it would be worth adding an extra layer to the file current _gpkg_ with Multi geometries and another one with Points to catch the errors ?! -> just confirming before I go ahead and change files and stuff. :) 
I appreciate the extra attention this is taking, btw!

Reply:
> As for the TypeError, do you mean that it would be worth adding an extra layer to the file current gpkg with Multi geometries and another one with Points to catch the errors ?! -> just confirming before I go ahead and change files and stuff. :)

Not worth it.